src/components/Article/ArticlesCurrentUser.jsx

```diff
@@ -0,0 +1,204 @@
+                data-index={index}
+                onClick={this.openModal}
+              >
+                <i className="fa fa-trash" data-index={index} />
+              </Button>
+            </div>
+            <p className="my-article__text">{article.description || article.body}</p>
+            <p className="my-article__meta">{this.renderDate(article.createdAt)}</p>
+          </div>
+        );
+      }
+    });
+  };
+
+  render() {
+    const {
+      articleStatus, showToast, deleteStatus, deleteMessage,
+    } = this.state;
+    return (
+      <section className="main-content">
+        <div className="container">
+          <Toast show={showToast} type={deleteStatus} message={deleteMessage} />
+          {this.renderModal()}
+          <div className="row content-margin">
+            <div className="col-1" />
+            <div className="col-10 mt-10">
+              <div className="row">
+                <div className="col-6-mob">
+                  <h3>My Articles</h3>
+                </div>
+                <div className="col-6-mob align-right">
+                  <h3>
+                    <Link to="/articles/create">New Article</Link>
+                  </h3>
+                </div>
+              </div>
+              <div className="my-articles-tab">
+                <span
+                  data-status="unpublished"
+                  role="presentation"
+                  className={articleStatus === 'unpublished' ? 'active' : ''}
+                  onClick={this.toggleArticleStatus}
+                >
+                  Draft
+                </span>
+                <span
+                  data-status="published"
+                  role="presentation"
+                  className={articleStatus === 'published' ? 'active' : ''}
+                  onClick={this.toggleArticleStatus}
+                >
+                  Published
+                </span>
+              </div>
+              <div className="my-articles row">
+                <div className="col-12">{this.renderArticles()}</div>
+              </div>
+            </div>
+          </div>
+        </div>
+        <a className="go-top-btn" href="#">
+          <i className="fa fa-angle-up" />
+        </a>
+      </section>
+    );
+  }
+}
+
+export const mapStateToProps = ({
+  currentUser: {
+    message, following, profile, deletingArticle, articles,
+  },
+}) => ({
+  message,
+  deletingArticle,
+  currentUser: profile,
+  following,
+  articles,
+});
+
+export const mapDispatchToProps = dispatch => ({
+  deleteArticle: payload => dispatch(onUserDeleteArticle(payload)),
+});
+
+ArticlesCurrentUser.propTypes = {
+  deletingArticle: PropTypes.bool,
+  currentUser: PropTypes.object,
+  deleteArticle: PropTypes.func.isRequired,
+};
+
+ArticlesCurrentUser.defaultProps = {
+  deletingArticle: false,
+  currentUser: {
+    articles: [],
+  },
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps,
+)(ArticlesCurrentUser);
```

Comment:
@mcaleb808 @filschristian @Nziranziza? You are all going to pay for this

Reply:
I am not sure I understand what you mean? Can you help us understand it better?